package.json

```diff
@@ -16,27 +16,32 @@
     "sort-imports": "eslint --fix .",
     "test": "jest",
     "test:watch": "jest --watch",
+    "build:btcwallet:mainnet": "webpack --config webpack.btcWallet.config.js --env NEXT_PUBLIC_NETWORK=mainnet",
```

Comment:
This looks weird. Instead of putting everything into our DApp application we could create a separate repo for test wallet and deploy it to npm. We could create a very simple browser extension for test purposes and inject it as devDependency into our main application.

I like the Tomo way of doing thing, they have created a separate repo even for their (our) wallet [providers](https://github.com/UnyxTech/tomo-wallet-provider), what makes much easier to maintain their SDK and Wallet Providers

Reply:
Yep, as I mentioned on the call, we now have `simple-staking` and a lot of stuff to build the wallet for e2e purposes.
Question though is are we ready to extract wallet providers and all of that outside of simple-staking? Is it a good time **now** or should we do it later?

Reply:
All this stuff around wallet providers is so confusing for me right now... If we have mobile/injectable provider in our app why just not use it for the test wallet? We can implement this interface in a separate repo and manually inject it into global window object before we run tests

Reply:
If I'm getting things right here is the list of packages and modules which we need extract to a new repo

`import "core-js/web/url";

import * as ecc from "@bitcoin-js/tiny-secp256k1-asmjs";
import * as bip32 from "@scure/bip32";
import * as bip39 from "@scure/bip39";
import * as bitcoin from "bitcoinjs-lib";
import ECPairFactory from "ecpair";

import { getNetworkFees, getTipHeight } from "@/utils/mempool_api";
import { getPublicKeyNoCoord, isTaproot } from "@/utils/wallet";
import {
  Fees,
  InscriptionIdentifier,
  Network,
  UTXO,
  WalletProvider,
} from "@/utils/wallet/wallet_provider";

import { nativeSegwitMainnetUTXOs } from "../mock/mainnet/nativeSegwit/utxos";
import { taprootMainnetUTXOs } from "../mock/mainnet/taproot/utxos";
import { nativeSegwitSignetUTXOs } from "../mock/signet/nativeSegwit/utxos";
import { taprootSignetUTXOs } from "../mock/signet/taproot/utxos";`

Reply:
> If we have mobile/injectable provider in our app why just not use it for the test wallet
I did not find the solution for the wallet to be "as it is" in typescript code and we are just injecting the stuff into window.btcwallet that works in node.js and Chrome (other browsers) Playwright env. As soon as we are starting to say "bitcoinjs sign this" or "bip39/32 get the private key" or "mempool fetch the data" it will cause problems
The only solution to this issue that I found is to build a JS wallets packages (build) that have everything they need. Then playwright injects it and calls needed methods